forte/data/data_store.py

```diff
@@ -1212,11 +1215,58 @@ def co_iterator_annotation_like(
 
         Args:
             type_names: a list of string type names
+            range_annotation: a list with two elements,
+                the start and the end of the range in
+                which we want to get required entries
 
         Returns:
             An iterator of entry elements.
         """
 
+        def get_bisect_range(entry_class: str, search_list: SortedList):
+            """
+            Perform binary search on the specified list for target entry class.
+            Args:
+                entry_class: Name of the target type of entry. It can be Annotation or
+                    `AudioAnnotation`.
+                search_list: A `SortedList` object on which the binary search
+                    will be carried out.
+            """
+            range_begin = (
+                range_annotation[constants.BEGIN_INDEX]
+                if range_annotation
+                else 0
+            )
+            range_end = (
+                range_annotation[constants.END_INDEX]
+                if range_annotation
+                else search_list[-1][constants.END_INDEX]
+            )
+
+            base_class = (
+                Annotation
+                if self._is_subclass(entry_class, Annotation)
+                else AudioAnnotation
+            )
+
+            temp_begin = self.add_entry_raw(
```

Comment:
`SortedList` can perform binary searching based on [sorting function](https://github.com/asyml/forte/blob/62b081ade30b9ab478d15b70954d5fead2029aab/forte/data/data_store.py#L764), so maybe we don't even need to create these temporary entries for index searching.